enumerable.rb

```diff
@@ -34,74 +34,100 @@ def my_select
     new_arr
   end
 
-  def my_all?
-    return self unless block_given?
+  def my_checker(cond)
+    return false if cond.nil? || cond == false
 
-    w = 0
-    while w < length
-      return false unless yield(self[w])
+    true
+  end
 
-      w += 1
+  def my_all?(_arg = nil)
+    if block_given?
+      w = 0
+      while w < length
+        return false unless yield(self[w])
+
+        w += 1
+      end
+      return true
+    end
+    my_each do |x|
+      return false if my_checker(x) == false
     end
     true
   end
 
-  def my_any?
-    return self unless block_given?
-
-    i = 0
-    while i < length
-      return true if yield(self[i])
+  def my_any?(_arg = nil)
+    if block_given?
+      i = 0
+      while i < length
+        return true if yield(self[i])
 
-      i += 1
+        i += 1
+      end
+      return false
+    else
+      (0...length).each do |x|
+        return true if !self[x].nil? && self[x] != false
+      end
     end
     false
   end
```

Comment:
### These tests are failing in `my_any?` method
- enumerables my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class
- enumerables my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex
- enumerables my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern